packages/flutter_tools/lib/src/project.dart

```diff
@@ -544,7 +544,7 @@ class AndroidProject extends FlutterProjectPlatform {
         // pluginManagement block of the settings.gradle file.
         // See https://docs.gradle.org/current/userguide/composite_builds.html#included_plugin_builds,
         // as well as the settings.gradle and build.gradle templates.
-        final bool declarativeApply = line.contains('dev.flutter.flutter-gradle-plugin');
+        final bool declarativeApply = line.contains(RegExp('dev\\.flutter\\.(?:(?:flutter-gradle-plugin)|(?:`flutter-gradle-plugin`))'));
```

Comment:
Use a raw string to avoid escape sequences when possible (this should fix the failing "Linux analyze" check). Also, move the argument to its own line to stay closer to our target maximum-line length of 80.

```suggestion
        final bool declarativeApply = line.contains(
          RegExp(r'dev\.flutter\.(?:(?:flutter-gradle-plugin)|(?:`flutter-gradle-plugin`))'),
        );
```